src/oidcc_token.erl

```diff
@@ -232,7 +232,13 @@ has_other_audience(ClientId, Audience) when is_list(Audience) ->
 
 
 validate_signature(IdToken, Kid, PubKeys, ProviderId) ->
-    PubKey = get_needed_key(PubKeys, Kid),
+    PubKeys1 =
+        case PubKeys of
+            [] -> refetch_keys(ProviderId);
+            _ -> PubKeys
+        end,
+
```

Comment:
this is not needed, or at least should not be needed, as the error on line 244 is only thrown in case ProviderId is undefined, which is after re-fetching the keys which is synchronized (waiting for the keys to come back).